log/truncating_file_writer.go

```diff
@@ -38,15 +41,36 @@ func NewTruncatingFileWriter(f *os.File, intervals ...time.Duration) TruncatingF
 }
 
 func (w *truncatingFileWriter) Write(p []byte) (n int, err error) {
-	if w.interval.Nanoseconds() > 0 && (time.Now().UnixNano()-w.lastTruncated.UnixNano() >= w.interval.Nanoseconds()) {
-		if err := w.Truncate(); err != nil {
-			w.lastTruncated = time.Now()
-		}
+	now := time.Now()
+	lastTruncated := w.getLastTruncatedUnixNano()
+	if w.interval.Nanoseconds() > 0 && (shouldTruncate(now, lastTruncated, w.interval)) {
```

Comment:
` w.interval.Nanoseconds() > 0` is unnecessary as it is being checked inside `shouldTruncate`